### PR TITLE
[docker, doc] feat: add ROCm Dockerfile and installation guide

### PR DESCRIPTION
**Pull request by Vivicai1005**
### What does this PR do?

Adds AMD GPU (ROCm) support: a `docker/Dockerfile.rocm` image for Instinct MI300X/MI350X, and a `docs/start/install_rocm.md` guide alongside the existing CUDA and Ascend NPU install pages.

### Checklist Before Starting

- [x] Search for similar PRs. Paste at least one query link here: ...
- [x] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `vllm_omni`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `diffusion`, `omni`, `tests`, `docker`
  - If this PR involves multiple modules, separate them with `,` like `[diffusion, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][diffusion, fsdp] feat: new rollout scheduler`

### Test

 ```bash
  bash examples/flowgrpo_trainer/qwen_image/run_qwen_image_ocr_lora.sh \
    actor_rollout_ref.model.attn_backend=native \
    actor_rollout_ref.rollout.rollout_attn_backend=TORCH_SDPA \
    trainer.total_training_steps=2
 ```
  Completes successfully (exit 0): 2 training steps at ~380 s/step,
  critic/rewards/mean: 0.765, throughput 0.334 img/s.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x]  Read the [Contribute Guide](https://github.com/verl-project/verl-omni/blob/main/CONTRIBUTING.md).
- [x]  Apply [pre-commit checks](https://github.com/verl-project/verl-omni/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [x] Add / Update the documentation.
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/verl-project/verl-omni/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...

### CODE REVIEW

File: docs/start/install_rocm.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+`PIP_CONSTRAINT` is baked into the image, so these cannot pull a CUDA `torch` over the ROCm build.
+
+## Run
+
+```bash
+docker run -it --rm \
+  --network host \
+  --ipc host \
+  --device=/dev/kfd \
```
Comment (wtomin) [inline docs/start/install_rocm.md:71]: [vLLM official ROCm image](https://docs.vllm.ai/en/stable/deployment/docker/#amd-rocm) also requires `--cap-add=SYS_PTRACE` (HIP JIT). verl `amd_quick_start` adds `--privileged`, `--ulimit memlock=-1 --ulimit stack=67108864`, and host ROCm **7.14**. PR Requirements say ROCm ≥ 7.0 with no host-driver section.

Are these misalignments critical?

File: docs/start/install_rocm.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+whose agent loop reads a `data.continuous_token` key the diffusion trainer
+config does not define.
+
+## Build Your Own Docker Image
+
+* AMD GPU (ROCm) Dockerfile: [`docker/Dockerfile.rocm`](https://github.com/verl-project/verl-omni/blob/main/docker/Dockerfile.rocm)
+
+NVIDIA GPU images are documented in the {doc}`GPU installation guide <install>`, and Ascend NPU images in the {doc}`NPU installation guide <install_npu>`.
+
+## Example: Qwen-Image FlowGRPO training in Docker
```
Comment (wtomin) [inline docs/start/install_rocm.md:171]: `docker/Dockerfile.rocm` COPY only pyproject, setup.py, README, pins, and `verl_omni/`. Same as CUDA. 

CUDA `install.md` therefore mounts `"$REPO:/workspace/verl-omni"`, `$WORKSPACE/data`, checkpoints, and `~/.cache/huggingface`. ROCm Run + Example still tell you to `python3 examples/flowgrpo_trainer/...` inside a container that does not contain `examples/`.

File: docs/start/install_rocm.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+This walkthrough uses the OCR dataset and
+`examples/flowgrpo_trainer/qwen_image/run_qwen_image_ocr_lora.sh`. It needs 8
```
Comment (wtomin) [inline docs/start/install_rocm.md]: OCR walkthrough says 8 GPUs (4 actor+rollout, 4 reward), which is wrong. Script uses n_gpus_per_node=4, REWARD_TP=4, colocated.

File: docker/Dockerfile.rocm [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+ && pip install --no-deps \
+      aenum==3.1.16 \
+      janus \
+      torchsde \
+      x-transformers \
+      cache-dit==1.5.0 \
+      onnxruntime \
+      prettytable \
+      gguf \
+      kernels==0.14.1 \
```
Comment (wtomin) [inline docker/Dockerfile.rocm]: The required `kernels` verison is `0.16.0`.

**Comment by wtomin** [comment]
@Vivicai1005 Thank you! I have a few comments. The rest looks fine to me.

**Comment by SamitHuang** [comment]
Are the installation and training recipes for different models same as Nvidia GPUs?

**Comment by Vivicai1005** [comment]
> @Vivicai1005 Thank you! I have a few comments. The rest looks fine to me.

Hi @wtomin Thanks for the comments! I’ve updated the PR based on your feedback.

**Comment by Vivicai1005** [comment]
> Are the installation and training recipes for different models same as Nvidia GPUs?

Hi @SamitHuang, the training recipes are shared with NVIDIA GPUs, but the installation/runtime setup is ROCm-specific. For diffusion recipes on ROCm, the main difference is that the attention backends need to be overridden to native / TORCH_SDPA.

File: docs/start/install_rocm.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+
+A few choices in `Dockerfile.rocm` differ from
+[`docker/Dockerfile.cuda`](https://github.com/verl-project/verl-omni/blob/main/docker/Dockerfile.cuda)
+and are worth knowing before you modify it.
+
+**vLLM is compiled from source, before anything else is installed.** There are
+no ROCm wheels on PyPI, so `pip install vllm==0.28.0` cannot work. Building it
+first keeps the compile environment identical to the one the base image
+validated — notably, installing `kernels` without `kernels-data` registers a
+setuptools entry point that breaks every subsequent source build in the image.
```
Comment (zhtmike) [inline docs/start/install_rocm.md]: no need to mention the debugging things here.

File: docs/start/install_rocm.md [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+
+**Two dependency corrections are applied** that the ROCm base needs:
+`diffusers` is raised to 0.40.0 (0.33.1 references `FLAX_WEIGHTS_NAME`, removed
+in transformers v5), and `torchao` to >=0.16.0 (`peft` >=0.19 raises
+`ImportError` when an older torchao is present, even though these recipes never
+use torchao quantization).
+
+**`verl` is reinstalled at the repo pin.** The base image ships 0.9.0.dev0,
+whose agent loop reads a `data.continuous_token` key the diffusion trainer
+config does not define.
```
Comment (zhtmike) [inline docs/start/install_rocm.md]: same; keep debugging content simple.

And drop version-specific things, it will be broken after next pin upgrade.

**Review by zhtmike — COMMENTED** [review]
Have you ever tried with training with qwen-image using ROCm, is the result normal?

**Comment by Vivicai1005** [comment]
> Have you ever tried with training with qwen-image using ROCm, is the result normal?

Hi @zhtmike Yes, this is the result from my 100-step GRPO run on ROCm.

image

**Comment by Vivicai1005** [comment]
> Have you ever tried with training with qwen-image using ROCm, is the result normal?

Hi @zhtmike yes, this is the result I got:

Test
 bash examples/flowgrpo_trainer/qwen_image/run_qwen_image_ocr_lora.sh \
   actor_rollout_ref.model.attn_backend=native \
   actor_rollout_ref.rollout.rollout_attn_backend=TORCH_SDPA \
   trainer.total_training_steps=2

Completes successfully (exit 0): 2 training steps at ~380 s/step,
critic/rewards/mean: 0.765, throughput 0.334 img/s.

**Comment by zhtmike** [comment]
I am fine with this PR, @wtomin pls check~

**Review by zhtmike — APPROVED** [review]

**Review by wtomin — APPROVED** [review]
LGTM.
